# Kids First DRC Consensus Calling Workflow
This workflow is used by the Kids First (KF) Data Resource Center (DRC) to create consensus calls from outputs generated by our somatic variant callers.

![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)

This workflow takes the protected vcf outputs from the [Kids First DRC Somatic Workflow](workflow/kfdrc-somatic-variant-workflow.cwl) and creates protected and public consensus VCF and MAF files.
The general outline is as follows:

1. Prep MNP Variants
   - Strelka2 outputs multi-nucleotide polymorphisms (MNPs) as consecutive single-nucleotide polymorphisms
   - In order preserve MNPs, we gather MNP calls from the other caller inputs, and search for evidence supporting these consecutive SNP calls as MNP candidates
    - Once found, the Strelka2 SNP calls supporting a MNP are converted to a single MNP call
    - This is done to preserve the predicted gene model as accurately as possible in our consensus calls
1. Consensus merge
   - Calls are gathered from all four callers
   - By default, calls with support from 2+ callers OR calls that are marked as `HotSpotAllele` in the `INFO` field are retained
   - Retained calls then have their `MQ` and `MQ0` values calculated from the input tumor cram
   - `GT` fields are estimated as "majority rules," and when no majority exists, set as `0/1` by default
   - `AD`, `DP`, and `AF` are calculated as the average value between callers
   - `ADR`, `DPR`, and `AFR` fields are added as the range of values from the previous point, to give the observer a sense on confidence in the value
1. VEP Annotate Consensus
1. BCF tools annotate
   - Additional annotation is performed augment VEP annotation
   - While VEP does have extensive gnomad allele frequency annotation, it is limited to exome values. The added gnomad AF only resource we use augments this as an additional `INFO/AF` field to add WGS frequencies
1. Soft filter variants
   - A soft filter is added based on criteria provided
   - By default, we perform soft filtering as outlined in the [KFDRC Annotation Subworkflow](kfdrc_annotation_subworkflow.md#workflow_description_and_kf_recommended_inputs)
1. VCF2MAF protected
   - Here, for convenience of analysis we convert the resultant, soft-filtered VCF (AKA, "Protected VCF") into MAF format
1. Hard filter VCF
   - The Protected VCF is hard filtered on `PASS` and `HotSpotAllele` for reasons outlined in the `Soft filter variants` step
   - This VCF is known as the "Public VCF"
1. VCF2MAF public
1. Rename outputs

![Consensus workflow](kfdrc-consensus-calling.png)

## Workflow Description and KF Recommended Inputs

### General workflow inputs:
```yaml
inputs:
  indexed_reference_fasta: {type: File, secondaryFiles: ['.fai', '^.dict'], 
                            sbg:suggestedValue: {class: File, path: 60639014357c3a53540ca7a3, name: Homo_sapiens_assembly38.fasta,
                            secondaryFiles: [{class: File, path: 60639016357c3a53540ca7af, name: Homo_sapiens_assembly38.fasta},
                            {class: File, path: 60639019357c3a53540ca7e7, name: Homo_sapiens_assembly38.dict}]
                            }}
  strelka2_vcf: {type: File, secondaryFiles: ['.tbi']}
  mutect2_vcf: {type: File, secondaryFiles: ['.tbi']}
  lancet_vcf: {type: File, secondaryFiles: ['.tbi']}
  vardict_vcf: {type: File, secondaryFiles: ['.tbi']}
  cram: {type: File, secondaryFiles: ['.crai'], doc: "Tumor cram recommended for MQ score calculation"}
  input_tumor_name: string
  input_normal_name: string
  output_basename: string
  ncallers: {type: int?, doc: "Optional number of callers required for consensus [2]", default: 2}
  annotation_vcf: {type: File, secondaryFiles: ['.tbi'], doc: "VCF of annotations to add to consensus variants, e.g. gnomAD allele frequency",
                   sbg:suggestedValue: {class: File, path: 5f50018fe4b054958bc8d2e3, name: af-only-gnomad.hg38.vcf.gz,
                   secondaryFiles: [{class: File, path: 5f50018fe4b054958bc8d2e5, name: af-only-gnomad.hg38.vcf.gz.tbi}]
                   }}
  vep_cache: {type: File, doc: "tar gzipped cache from ensembl/local converted cache",
    sbg:suggestedValue: {class: File, path: 607713829360f10e3982a425, name: homo_sapiens_vep_93_GRCh38.tar.gz}}
  annot_columns: {type: string?, default: 'INFO/AF', doc: "column from annotation_vcf to add to consensus VCF; defaults to 'INFO/AF'"}
  filter_names: {type: 'string[]?', default: [ "NORM_DP_LOW", "GNOMAD_AF_HIGH" ], doc: "Names of filters to be added to consensus VCF;\
     \ default values set"}
  depth_lowerbound: {type: int?, default: 7, doc: "Normal-sample read depth at which to apply depth filter; default set"}
  frequency_upperbound: {type: float?, default: 0.001, doc: "Population allele frequency above which to apply frequency filter; default set"}
  bcftools_public_filter: {type: string?, doc: 'Will hard filter final result to create a public version, e.g. FILTER="PASS"|INFO/HotSpotAllele=1; default set', default: 'FILTER="PASS"|INFO/HotSpotAllele=1'}
  retain_info: {type: string?, doc: "csv string with INFO fields that you want to keep; default values set", default: 'MQ,MQ0,CAL,HotSpotAllele'}
  retain_fmt: {type: string?, doc: "csv string with FORMAT fields that you want to keep"}
  maf_center: {type: string?, doc: "Sequencing center of variant called", default: "."}
```

### Recommended reference inputs - all file references can be obtained [here](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-references/)
Secondary files needed for each reference file will be a sub-bullet point
 - `indexed_reference_fasta`: `Homo_sapiens_assembly38.fasta`
   - `Homo_sapiens_assembly38.fasta.fai`
   - `Homo_sapiens_assembly38.dict`
 - `bcftools_annot_columns`: "INFO/AF"
 - `annotation_vcf`: `af-only-gnomad.hg38.vcf.gz`
   - `af-only-gnomad.hg38.vcf.gz.tbi`
 - `bcftools_public_filter`: 'FILTER="PASS"|INFO/HotSpotAllele=1'
 - `vep_cache`: `homo_sapiens_vep_93_GRCh38.tar.gz`

### Situational inputs
 - `depth_lowerbound`: Change this `int` if you believe the threshold for `NORM_DP_LOW` should be different
 - `frequency_upperbound` Change this `float` if you believe the max value for `GNOMAD_AF_HIGH` should be different
 - `ncallers`: Change this `int` to adjust the stringency of what is defined as a consensus

## Workflow outputs
```yaml
outputs:
  annotated_protected_outputs: {type: 'File[]', outputSource: rename_protected/renamed_files}
  annotated_public_outputs: {type: 'File[]', outputSource: rename_public/renamed_files}

```
 - `annotated_protected_outputs`: Array of files containing MAF format of PASS hits, `PASS` VCF with annotation pipeline soft `FILTER`-added values, and VCF index
 - `annotated_public_outputs`: Same as above, except MAF and VCF have had entries with soft `FILTER` values removed

# Simulated bash calls
For your convenience, we have exported the calls from an example workflow, as they were run in each docker image for each step:

 | Step                        | Type         | Num scatter            | Command                                                                                                                                                                                                         |
 | --------------------------- | ------------ | ---------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | prep_mnp_variants                        | run step         | NA            | bcftools view --types snps /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/e4765887-135d-4763-92a8-c149bcc0fb2e.strelka2_somatic.norm.annot.protected.vcf.gz -Oz > strelka2_snps.vcf.gz                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | bcftools view --types indels /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/e4765887-135d-4763-92a8-c149bcc0fb2e.strelka2_somatic.norm.annot.protected.vcf.gz -Oz > strelka2_indels.vcf.gz                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | tabix strelka2_indels.vcf.gz                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | echo creating mnp table;cp /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/ae99c014-245a-4820-b62b-a079f672a92d.mutect2_somatic.norm.annot.protected.vcf.gz 2_ae99c014-245a-4820-b62b-a079f672a92d.mutect2_somatic.norm.annot.protected.vcf.gz;cp /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/ae99c014-245a-4820-b62b-a079f672a92d.mutect2_somatic.norm.annot.protected.vcf.gz.tbi 2_ae99c014-245a-4820-b62b-a079f672a92d.mutect2_somatic.norm.annot.protected.vcf.gz.tbi;bcftools view --type mnps -H /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/ae99c014-245a-4820-b62b-a079f672a92d.mutect2_somatic.norm.annot.protected.vcf.gz >> mnps.vcf;cp /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/9f4fbdcb-025e-4d52-9fa6-0dff0b4b3740.lancet_somatic.norm.annot.protected.vcf.gz 3_9f4fbdcb-025e-4d52-9fa6-0dff0b4b3740.lancet_somatic.norm.annot.protected.vcf.gz;cp /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/9f4fbdcb-025e-4d52-9fa6-0dff0b4b3740.lancet_somatic.norm.annot.protected.vcf.gz.tbi 3_9f4fbdcb-025e-4d52-9fa6-0dff0b4b3740.lancet_somatic.norm.annot.protected.vcf.gz.tbi;bcftools view --type mnps -H /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/9f4fbdcb-025e-4d52-9fa6-0dff0b4b3740.lancet_somatic.norm.annot.protected.vcf.gz >> mnps.vcf;cp /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/c4561206-8dcd-48cc-919e-28a902f01ec0.vardict_somatic.norm.annot.protected.vcf.gz 4_c4561206-8dcd-48cc-919e-28a902f01ec0.vardict_somatic.norm.annot.protected.vcf.gz;cp /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/c4561206-8dcd-48cc-919e-28a902f01ec0.vardict_somatic.norm.annot.protected.vcf.gz.tbi 4_c4561206-8dcd-48cc-919e-28a902f01ec0.vardict_somatic.norm.annot.protected.vcf.gz.tbi;bcftools view --type mnps -H /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/c4561206-8dcd-48cc-919e-28a902f01ec0.vardict_somatic.norm.annot.protected.vcf.gz >> mnps.vcf;cut -f 1,2,4,5 mnps.vcf | sort -V -k 1,1 -k 2,2 | uniq > mnps_sorted.txt;                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | python3 -c "def main():                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     import sys                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     import gzip                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                          |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_fn = 'strelka2_snps.vcf.gz'                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_snps = gzip.open(strelka2_fn)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                          |
 | prep_mnp_variants                        | run step         | NA            |     # store mnps, with chrom\tpos as key, list of mnps as values                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     mnp_dict = {}                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     with open('mnps_sorted.txt', 'r') as mnps_txt:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         for line in mnps_txt:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             # use ASCII codes for \n and \t for compat. w/ cwltool + Cavatica                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             info = line.rstrip(chr(10)).split(chr(9))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             key = chr(9).join(info[:2])                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             if key not in mnp_dict:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 mnp_dict[key] = []                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             mnp_dict[key].append(chr(9).join(info[2:4]))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                          |
 | prep_mnp_variants                        | run step         | NA            |     header = []                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     # store header                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     for line in strelka2_snps:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         header.append(line.decode())                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         if line.decode().startswith('#CHROM'):                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             break                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     # on rescan, start at data skipping over header                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     last_pos = strelka2_snps.tell()                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     # use chrom\tpos as starting point to eval a possible overlapping mnp                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     new_mnps = {}                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                          |
 | prep_mnp_variants                        | run step         | NA            |     for line in strelka2_snps:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         info = line.decode().rstrip(chr(10)).split(chr(9))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         dkey = chr(9).join(info[:2])                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         if dkey in mnp_dict:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             max_len = len(max(mnp_dict[dkey], key=len))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             candidates = {}                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             # if possible, build longest supported mnp                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             ref = info[3]                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             alt = info[4]                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             cur = ref + chr(9) + alt                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             pos = info[1]                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             for val in mnp_dict[dkey]:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 candidates[val] = 0                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             try:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 f = 0                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 c = 0                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 while f == 0:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     # keep track of current search position in file, if condition met where the next position isn't continous, or exceeds the longest seen mnp, pick up from there in the loop                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     cur_pos = strelka2_snps.tell()                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     new_in = next(strelka2_snps)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     new_info = new_in.decode().rstrip(chr(10)).split(chr(9))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     if new_info[0] == info[0] and int(new_info[1]) == (int(pos) + 1):                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         ref += new_info[3]                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         alt += new_info[4]                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         cur = ref + chr(9) + alt                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         pos = new_info[1]                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         if cur in candidates:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                             candidates[cur] = 1                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                             c = 1                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         if len(cur) == max_len:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                             f = 1                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                             strelka2_snps.seek(cur_pos)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     else:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         f = 1                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                         strelka2_snps.seek(cur_pos)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 if c == 1:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     new_mnps[dkey] = cur                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     sys.stderr.write('Candidate mnp found: ' + dkey + chr(9) + cur + chr(10))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                                                  |
 | prep_mnp_variants                        | run step         | NA            |             except Exception as e:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 sys.stderr.write(str(e) + chr(10))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 sys.stderr.write('Probably hit EOF' + chr(10))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 if c == 1:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                     new_mnps[dkey] = cur                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_snps.close()                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_modded = open('strelka2_snp_mnp.vcf', 'w')                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_modded.write(''.join(header))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_snps = gzip.open(strelka2_fn)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_snps.seek(last_pos)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                          |
 | prep_mnp_variants                        | run step         | NA            |     # re-scan strelka2 snps, if position in candidate mnps, use that info, else output snp info                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     for line in strelka2_snps:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         info = line.decode().rstrip(chr(10)).split(chr(9))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         dkey = chr(9).join(info[:2])                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         if dkey in new_mnps:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             (ref, alt) = new_mnps[dkey].split(chr(9))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             strelka2_modded.write(chr(9).join([dkey, info[2], new_mnps[dkey]] + info[5:]) + chr(10))                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             for i in range(1, len(alt), 1):                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                 skip = next(strelka2_snps)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |         else:                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |             strelka2_modded.write(line.decode())                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     strelka2_modded.close()                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     sys.exit(0)                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | if __name__ == '__main__':                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |     main()"                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            |                                                                                                                                                                                                          |
 | prep_mnp_variants                        | run step         | NA            | bgzip strelka2_snp_mnp.vcf && tabix strelka2_snp_mnp.vcf.gz                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | bcftools concat -a strelka2_snp_mnp.vcf.gz strelka2_indels.vcf.gz -O z -o 1_DELME_TEST.strelka2.mnp_phased.vcf.gz                                                                                                                                                                                                         |
 | prep_mnp_variants                        | run step         | NA            | tabix 1_DELME_TEST.strelka2.mnp_phased.vcf.gz                                                                                                                                                                                                         |
 | consensus_merge                        | run step         | NA            | /usr/bin/consensus_merge.py --strelka2_vcf /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/prep_mnp_variants/1_DELME_TEST.strelka2.mnp_phased.vcf.gz --mutect2_vcf /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/ae99c014-245a-4820-b62b-a079f672a92d.mutect2_somatic.norm.annot.protected.vcf.gz --lancet_vcf /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/9f4fbdcb-025e-4d52-9fa6-0dff0b4b3740.lancet_somatic.norm.annot.protected.vcf.gz --vardict_vcf /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/c4561206-8dcd-48cc-919e-28a902f01ec0.vardict_somatic.norm.annot.protected.vcf.gz --cram /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/DELME_ANNOT_TEST/c8ab11b5-bfd4-4842-a912-8be36855d72a.cram --reference /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/Homo_sapiens_assembly38.fasta --ncallers 2 --output_basename DELME_TEST                                                                                                                                                                                                         |
 | vep_annot_consensus                        | run step         | NA            | mkdir homo_sapiens && tar --use-compress-program="pigz -p 8" -xf /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/homo_sapiens_vep_93_GRCh38.tar.gz -C homo_sapiens && perl /ensembl-vep-release-93.7/vep --af --af_1kg --af_esp --af_gnomad --allele_number --assembly GRCh38 --biotype --buffer_size 10000 --cache --cache_version 93 --canonical --ccds --check_existing --dir_cache homo_sapiens --domains --failed 1 --fasta /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/Homo_sapiens_assembly38.fasta --flag_pick_allele --fork 16 --format vcf --gene_phenotype --hgvs --input_file /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/consensus_merge/DELME_TEST.vardict_somatic.norm.annot.protected.consensus.vcf.gz --no_escape --no_progress --no_stats  --numbers --offline --output_file DELME_TEST.consensus_somatic.PASS.vep.vcf --pick_order canonical,tsl,biotype,rank,ccds,length --polyphen b --protein --pubmed  --shift_hgvs 1 --sift b --species homo_sapiens --symbol --total_length --tsl --uniprot --variant_class --vcf --xref_refseq && /ensembl-vep-release-93.7/htslib/bgzip DELME_TEST.consensus_somatic.PASS.vep.vcf && /ensembl-vep-release-93.7/htslib/tabix DELME_TEST.consensus_somatic.PASS.vep.vcf.gz                                                                                                                                                                                                         |
 | bcftools_annotate                        | run step         | NA            | bcftools annotate --annotations /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/af-only-gnomad.hg38.vcf.gz --columns INFO/AF -o DELME_TEST.bcft_annot.bcf_annotated.vcf.gz -O z --threads 4 /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/vep_annot_consensus/DELME_TEST.consensus_somatic.PASS.vep.vcf.gz && tabix DELME_TEST.bcft_annot.bcf_annotated.vcf.gz                                                                                                                                                                                                         |
 | variant_filter                        | run step         | NA            | /gatk VariantFiltration -R /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/Homo_sapiens_assembly38.fasta -V /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/bcftools_annotate/DELME_TEST.bcft_annot.bcf_annotated.vcf.gz -O DELME_TEST.filter.gatk.soft_filtered.vcf.gz --filter-name "NORM_DP_LOW" --filter-expression "vc.getGenotype('BS_0KKH9VKP').getDP() <= 7" --filter-name "GNOMAD_AF_HIGH" --filter-expression "AF > 0.001"                                                                                                                                                                                                         |
 | hard_filter_vcf                        | run step         | NA            | bcftools view --include 'FILTER="PASS"|INFO/HotSpotAllele=1' /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/variant_filter/DELME_TEST.filter.gatk.soft_filtered.vcf.gz -O z > DELME_TEST.vcf.gz;tabix DELME_TEST.vcf.gz;                                                                                                                                                                                                         |
 | kfdrc_vcf2maf_protected                        | run step         | NA            | gunzip -c /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/variant_filter/DELME_TEST.filter.gatk.soft_filtered.vcf.gz > input_file.vcf && perl /vcf2maf/vcf2maf.pl --input-vcf input_file.vcf --output-maf DELME_TEST.protected.vep.maf --tumor-id BS_79SYEHY3 --normal-id BS_0KKH9VKP --ncbi-build GRCh38 --ref-fasta /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/Homo_sapiens_assembly38.fasta --maf-center "." --retain-info MQ,MQ0,CAL,HotSpotAllele                                                                                                                                                                                                         |
 | kfdrc_vcf2maf_public                        | run step         | NA            | gunzip -c /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/hard_filter_vcf/DELME_TEST.vcf.gz > input_file.vcf && perl /vcf2maf/vcf2maf.pl --input-vcf input_file.vcf --output-maf DELME_TEST.public.vep.maf --tumor-id BS_79SYEHY3 --normal-id BS_0KKH9VKP --ncbi-build GRCh38 --ref-fasta /sbgenomics/Projects/0154a615-2b20-4c32-9c76-141b5eebdde1/Homo_sapiens_assembly38.fasta --maf-center "." --retain-info MQ,MQ0,CAL,HotSpotAllele                                                                                                                                                                                                         |
 | rename_protected                        | run step         | NA            | mkdir renamed/;cp /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/variant_filter/DELME_TEST.filter.gatk.soft_filtered.vcf.gz renamed/DELME_TEST.protected.vcf.gz;cp /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/variant_filter/DELME_TEST.filter.gatk.soft_filtered.vcf.gz.tbi renamed/DELME_TEST.protected.vcf.gz.tbi;cp /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/kfdrc_vcf2maf_protected/DELME_TEST.protected.vep.maf renamed/DELME_TEST.protected.maf;                                                                                                                                                                                                         |
 | rename_public                        | run step         | NA            | mkdir renamed/;cp /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/hard_filter_vcf/DELME_TEST.vcf.gz renamed/DELME_TEST.public.vcf.gz;cp /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/hard_filter_vcf/DELME_TEST.vcf.gz.tbi renamed/DELME_TEST.public.vcf.gz.tbi;cp /sbgenomics/workspaces/0154a615-2b20-4c32-9c76-141b5eebdde1/tasks/0d982def-37a4-4dfe-8274-822cce0f5ac2/kfdrc_vcf2maf_public/DELME_TEST.public.vep.maf renamed/DELME_TEST.public.maf;                                                                                                                                                                                                         |
